### ISSUE TITLE
wk1 sep_epic: UAT Bug? Team - Current Session is off?

### ISSUE THREAD

**Issue by lijenn** [lzim/teampsd#2229]
1. **Who encountered the bug:** Jenn

2. **Which instance are you testing in?:** QA

3. **Date/Time of the bug:** 9/8 8:07am Pacific

4. **Role:** Admin

5. **Detailed description of the bug:** 
- Created a Live NIH MTL Red Team yesterday
- Set up Session Dates for Part I (9/8 at 8:30am) & Part II (9/22 at 12am)
- Set up due dates for Part I tasks and completed 1 out of 3 Part I tasks
- Sent Pre S01 email yesterday and received it after the nightly run
- Expected Current Session field on dashboard to show Part I, but it shows Part II
(Let me know if this is user error, I recall the Current Session being based off of the Session for the next upcoming or overdue Task?)

![image](https://user-images.githubusercontent.com/59668647/132536719-47b004ca-503a-4cb5-9356-83eeb81ab712.png)
![image](https://user-images.githubusercontent.com/59668647/132536875-8e6a3bab-2de3-4f15-9723-55da53b8fbe8.png)

**Comment by hsnyder2** [comment]
@lijenn  Yes, it looks like Current Session should be Part I, not Part II.  Forwarding to Arlene to troubleshoot.

**Comment by hsnyder2** [comment]
@lijenn ID'd that QA did not have the correct version of the plugin that runs this routine.  Arlene moved it to QA, ran it and the Current Session for this Team is now correct.  Ready for retesting.  Thx!

**Comment by lijenn** [comment]
Thanks @hsnyder2! I think the Current Session for both Coplanning and Teams are stuck now at the 1st Session after the nightly run?

LMK if it's because the checklist items don't have assigned due dates though? Just wanted to see it the Current Session would continue updating to the next Session based on the checklist items, if no due date was provided for the checklist item.

If no to the above, then I think this Group should be on Current Session 2A and this MTL Team should be on Current Session Part II:
![image](https://user-images.githubusercontent.com/59668647/132875998-c57a5a24-520a-44ee-861b-d9422fa87ed4.png)
![image](https://user-images.githubusercontent.com/59668647/132876204-32131035-84cf-4427-830d-c4233045a9bc.png)

**Comment by hsnyder2** [comment]
When did you complete all Team "Part I" tasks?  Same question for Group "1B" tasks?

**Comment by lijenn** [comment]
@hsnyder2 For both, I want to say yesterday roughly around 6-7pm Pacific.

**Comment by hsnyder2** [comment]
Hmm... Arlene just looked at some logs and if she's looking at the right records, is it possible that they were changed after 9:30pm PT?  The nightly job ran at that time (12:30pm ET) and she's seeing changes that were made at 9:45pm ET ... after the job ran.  She just reran the job that ran at 12:30 ET this a.m. and Current Sessions for those two are now correct.

**Comment by lijenn** [comment]
I honestly can't recall...but let me see if I understood the explanation correctly cause I think I'm missing something.

I think we changed the nightly run to start at 8AM ET/5AM PT the following day, right? 
And say I completed tasks yesterday between 9-1am ET/6-10pm PT the night before, wouldn't the Current Sessions have updated today around 8AM ET/5AM PT?

**Comment by hsnyder2** [comment]
I think that was just the job that sends emails, but let me check.  Might be quicker to resolve any follow on questions/issues on a call w Arlene but let me confirm answer to the timing question.

**Comment by lijenn** [comment]
Ohh, I was under the assumption that for the emails to go out at 8AM ET, the nightly plug in would have to start at the same time...If that's the case then I would say this is user error.

 I was testing it under the assumption that the nightly plug in had to run at 8AM ET the next day.

LMK if that's the case. If not, then I think it's fine for the nightly plug in to continue updating the Session at...I think it was 1am ET...right?

**Comment by hsnyder2** [comment]
Yeah, we like to schedule jobs in the wee hours to make sure they don't run into a lot of demand on the system. The times are easily changed, but as the volume of processing that they do gets bigger, the bigger the risk they'll run into issues if they're scheduled to run during business hours.  Arlene's confirmed that the current session job runs at 12:30AM ET, so, yeah, any changes you made after 9:30PM PT wouldn't have made it into that job.

**Comment by lijenn** [comment]
Got it! That makes total sense for the nightly plug in to run at 12:30 AM ET, I'll give this another test to confirm and double on the check emails sending out at the right time. Once those are good, I'll close this ticket!

**Comment by lijenn** [comment]
@hsnyder2 @apisano0920 Actually, related to emails. For this adhoc email that I was trying to send from D365 from the Emails under Main on the left hand nav, is this email pending because we're scheduling emails (all emails, not just templated ones?) to go out at 8AM ET? 

I clicked 'Send' on this email I drafted around 30 min ago.
![image](https://user-images.githubusercontent.com/59668647/132885329-0f6851b2-5225-461b-b9b9-ffb148390d4b.png)
